app/models/study.rb

```diff
@@ -1132,6 +1132,22 @@ def unique_genes
     Gene.where(study_id: self.id, :study_file_id.in => self.expression_matrix_files.map(&:id)).pluck(:name).uniq
   end
 
+  # Get species scientific name for a searched gene.  E.g. PTEN -> Homo sapiens.
+  def infer_species(searched_gene)
+    # Get gene record for searched gene string
+    gene = self.genes.by_name_or_id(searched_gene, self.expression_matrix_files.map(&:id))
```

Comment:
this is going to cause a double-search for the gene, since I assume this same search logic is happening for the expression viewer.  The performance hit doesn't seem to be large, but it might be worth consolidating the logic if possible just so there aren't bugs if the genes-by-study lookup logic changes

Reply:
The scope `by_name_or_id` is really only intended for merging data from multiple matrices into a single `Hash` for visualization purposes.  If the intent of this method is purely to return the scientific name of the taxon that a given `Gene` maps to, I agree with Devon that this is unnecessary overhead.  We already search for genes at least twice when doing gene expression views, so adding a third is not going to help with overall render times.

Reply:
I've updated the query per Jon's other comment; see [here](https://github.com/broadinstitute/single_cell_portal_core/blob/1731b32e9d0f2a7fbd6d89c177b5eebc95ca0641/app/models/study.rb#L1135).